features/294-login-without-specifying-login-id-key/api.md

```diff
@@ -0,0 +1,104 @@
+For supporting login without specifying `loginID`, have two proposals:
+
+## Proposal 1 - auth gear find corresponding user object automatically
+
+Keeping most of current implementation, auth gear finds possible user object automatically. Consider following use case:
+
+```javascript
+skygear.auth.signup({
+  email: 'example@example.com', 
+  username: 'example'
+}, 'password');
+
+// new:
+skygear.auth.login('example@example.com', 'password')
+.then((user) => {
+  console.log('login successfully');
+});
+```
+
+
+### Possible Problem
+
+If user signup with incorrect `loginIDMetadataKeys`, it may cause user identify miss assign. Consider following scenario:
+
+```javascript
+
+// signup user A
+skygear.auth.signup({
```

Comment:
I think the example given here is incorrect. In this example:

User A should be only able to sign in with `{email: 'example@example.com', username: 'example', 'password'}`, while User B only can sign in with  `{email: 'example', username: 'example@example.com', 'password'}`

There are no ambiguous. 

I think you want to give an example like:

```
skygear.auth.signup({email: 'example@example.com'}, 'password');
skygear.auth.signup({username: 'example@example.com'}, 'password');
```

In the above case `skygear.auth.login('example@example.com', 'password');` will have ambigous result.

Reply:
By current default setting `LOGIN_ID_METADATA_KEYS=[['username'], ['email']]`, `signup({email: 'example@example.com', username: 'example', 'password'})` and `signup({email: 'example', username: 'example@example.com', 'password'})` will become 4 principals:

`{email: 'example@example.com}`
`{username: 'example'}`
`{email: 'example'}`
`{username: 'example@example.com'}`

And a user could use one of them to login. So the example given here is ambiguous too. For demonstration purpose, I had updated it to a simpler version. Thanks.